### PR TITLE
{WIP} Add more information on where to find various logs.

### PR DESCRIPTION
**Pull request by ehiggs**
Some users are left confused where all the logging is done so they can figure out errors. This is an attempt to provide more information.

### CODE REVIEW

File: docs/Logging.rst [hunk 1]
```diff
@@ -1,6 +1,31 @@
+=======
+Logging
+=======
+If your job didn't work as expected, you'll need to check the logs. There are four places to consider:
+
+   1. When submitting your job to start the cluster, hanythingondemand logs to stdout. The potential errors here are usually:
```
Comment (boegel) [inline docs/Logging.rst]: s/stdout/your terminal session/g?

File: docs/Logging.rst [hunk 2]
```diff
@@ -1,6 +1,31 @@
...
+         1. PBS isn't running or isn't accessible. If so, contact your administrators.
+
+         2. Your envirnoment is broken. e.g. if you're using a Python version for a cluster that doesn't work on the login node.
+
+   2. If PBS is accessible and tries to run the job but it failed to start properly (e.g. due to a problem with MPI) you will see errors in ``Hanythingondemand.e$(jobid)``.
```
Comment (boegel) [inline docs/Logging.rst]: use `${PBS_JOBID}`?

Comment (boegel) [inline docs/Logging.rst]: Also, mention where this file can be found: directory where the job was submitted from

File: docs/Logging.rst [hunk 3]
```diff
@@ -1,6 +1,31 @@
...
+
+   3. When PBS starts your job, it will start logging to ``hod.output.$(hostname).$(pid)``. If your service configuration files have problems (e.g. typos in the commands, bad paths, etc) then the erorr will be here. For example if a service failed to start you will see a message in the logs saying: ``Problem occured with cmd``.
```
Comment (boegel) [inline docs/Logging.rst]: break up long lines (120 chars max)

mention where this file can be found

Comment (boegel) [inline docs/Logging.rst]: replace cmd with `<command>`

File: docs/Logging.rst [hunk 4]
```diff
@@ -1,6 +1,31 @@
...
+
+   4. The logs of the services running under the cluster. These are specific to
```
Comment (boegel) [inline docs/Logging.rst]: where are these? `--workdir`?

File: docs/Logging.rst [hunk 5]
```diff
@@ -1,6 +1,31 @@
...
+         the services being used.
+
 ==============
 Hadoop Logging
 ==============
 By default, the log files for a Hadoop cluster will be in ``$localworkdir/log``. Expanded, this is in the workdir as follows: ``$workdir/${USER}.${HOSTNAME}.${PID}/log``
 
 One of the advantages of having the log files on a parallel file system is that one no longer needs to use special tools for log aggregation (Flume, Logstash, Logly, etc) since all the logs for all the nodes are in a single directory structure.
```
Comment (boegel) [inline docs/Logging.rst]: break long lines pls

File: docs/Logging.rst [hunk 6]
```diff
@@ -1,6 +1,31 @@
...
+
+Hadoop logs have three components:
```
Comment (boegel) [inline docs/Logging.rst]: two or three?

File: docs/Logging.rst [hunk 7]
```diff
@@ -1,6 +1,31 @@
...
+
+    1. Service logs. These are in ``$localworkdir/log``. e.g. ``yarn-username-resourcemanager-node.domain.out``, ``yarn-username-nodemanager-node.domain.out``.
```
Comment (boegel) [inline docs/Logging.rst]: s/e.g./Examples are: /

File: docs/Logging.rst [hunk 8]
```diff
@@ -1,6 +1,31 @@
...
+
+    2. Container logs. Each piece of Hadoop work takes place in a container. Output from your program to stdout will appear in these files.  These are organized by application/container/stderr and stdout. e.g.: ::
```
Comment (boegel) [inline docs/Logging.rst]: s/to stdout//

Comment (boegel) [inline docs/Logging.rst]: s/e.g./For example: /

File: docs/Logging.rst (hunk 3 above)
Comment (ehiggs) [inline docs/Logging.rst]: cmd is in the literal string.

File: docs/Logging.rst (hunk 4 above)
Comment (ehiggs) [inline docs/Logging.rst]: Depends on the configuration. It's possible they log to /tmp on each node. This is why there is a section for Hadoop to explain where hadoop logs are.

File: docs/Logging.rst [hunk 9]
```diff
@@ -1,6 +1,48 @@
...
+         where you ran the job.
+
+   3. When PBS starts your job, it will start logging to
+         ``hod.output.$(hostname).$(pid)``. If your service configuration files
+         have problems (e.g. typos in the commands, bad paths, etc) then the
+         erorr will be here. For example if a service failed to start you will
+         see a message in the logs saying: ``Problem occured with cmd``.
+
+   4. The logs of the services running under the cluster. These are specific to
+         the services being used.
```
Comment (boegel) [inline docs/Logging.rst]: create a separate section `HOD services logging` (or something like that), refer to it from here (yes, I know it'll be right below this), and make `Hadoop` a subsection of it.

File: docs/Logging.rst [hunk 10]
```diff
@@ -1,6 +1,48 @@
-==============
-Hadoop Logging
-==============
-By default, the log files for a Hadoop cluster will be in ``$localworkdir/log``. Expanded, this is in the workdir as follows: ``$workdir/${USER}.${HOSTNAME}.${PID}/log``
+=======
+Logging
```
Comment (boegel) [inline docs/Logging.rst]: clarify that this section applies to `hanythingondemand` itself, not the services it is running

File: docs/Logging.rst [hunk 11]
```diff
@@ -1,6 +1,48 @@
...
+   3. When PBS starts your job, it will start logging to
+         ``hod.output.$(hostname).$(pid)``. If your service configuration files
+         have problems (e.g. typos in the commands, bad paths, etc) then the
+         erorr will be here. For example if a service failed to start you will
+         see a message in the logs saying: ``Problem occured with cmd``.
+
+   4. The logs of the services running under the cluster. These are specific to
+         the services being used.
+
+============== Hadoop Logging ==============
```
Comment (boegel) [inline docs/Logging.rst]: does this render correctly? use http://rst.ninjs.org/ to check

File: docs/Logging.rst [hunk 12]
```diff
@@ -1,6 +1,48 @@
...
-By default, the log files for a Hadoop cluster will be in ``$localworkdir/log``. Expanded, this is in the workdir as follows: ``$workdir/${USER}.${HOSTNAME}.${PID}/log``
+=======
+Logging
+=======
+If your job didn't work as expected, you'll need to check the logs. There are
+four places to consider:
 
-One of the advantages of having the log files on a parallel file system is that one no longer needs to use special tools for log aggregation (Flume, Logstash, Logly, etc) since all the logs for all the nodes are in a single directory structure.
+   1. When submitting your job to start the cluster, hanythingondemand logs to
+         your terminal session. The potential errors here are usually: 1. PBS
```
Comment (boegel) [inline docs/Logging.rst]: this won't render correctly as a list, I suspect, make sure it does

**Comment by boegel** [comment]
Going in, much improved, thanks @ehiggs!

**Pull request merged**